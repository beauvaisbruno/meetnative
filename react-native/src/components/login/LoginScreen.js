"use strict";
var __importStar =
  (this && this.__importStar) ||
  function(mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null)
      for (var k in mod)
        if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
  };
var __importDefault =
  (this && this.__importDefault) ||
  function(mod) {
    return mod && mod.__esModule ? mod : { default: mod };
  };
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importStar(require("react"));
const native_base_1 = require("native-base");
const FbLoginButton_1 = __importDefault(require("./FbLoginButton"));
const EmailLoginUi_1 = __importDefault(require("./EmailLoginUi"));
const GoogleLoginButton_1 = __importDefault(require("./GoogleLoginButton"));
const react_native_responsive_screen_1 = require("react-native-responsive-screen");
const react_native_auto_height_image_1 = __importDefault(
  require("react-native-auto-height-image")
);
class LoginScreen extends react_1.Component {
  render() {
    return react_1.default.createElement(
      native_base_1.Container,
      null,
      react_1.default.createElement(
        native_base_1.Content,
        null,
        react_1.default.createElement(
          react_native_auto_height_image_1.default,
          {
            source: require("./img/header.png"),
            width: react_native_responsive_screen_1.widthPercentageToDP("100%")
          }
        ),
        react_1.default.createElement(
          native_base_1.View,
          { padder: true },
          react_1.default.createElement(GoogleLoginButton_1.default, null),
          react_1.default.createElement(FbLoginButton_1.default, null),
          react_1.default.createElement(EmailLoginUi_1.default, null)
        )
      )
    );
  }
}
exports.default = LoginScreen;
