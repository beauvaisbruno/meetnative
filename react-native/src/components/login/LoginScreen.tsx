import React, { Component } from "react";
import { Container, Content, Text, View } from "native-base";
import FbLoginButton from "./FbLoginButton";
import EmailLoginUi from "./EmailLoginUi";
import GoogleLoginButton from "./GoogleLoginButton";
import { widthPercentageToDP as wp } from "react-native-responsive-screen";
import AutoHeightImage from "react-native-auto-height-image";

class LoginScreen extends Component {
  render() {
    return (
      <Container>
        <Content>
          <AutoHeightImage
            source={require("./img/header.png")}
            width={wp("100%")}
          />
          <View padder>
            <GoogleLoginButton />
            <FbLoginButton />
            <EmailLoginUi />
          </View>
        </Content>
      </Container>
    );
  }
}

export default LoginScreen;
