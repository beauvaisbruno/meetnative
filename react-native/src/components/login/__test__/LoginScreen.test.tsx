import { shallow } from "enzyme";
import * as React from "react";
import toJson from "enzyme-to-json";
import LoginScreen from "../LoginScreen";

it("Logout Button", () => {
  const tree = shallow(<LoginScreen />);
  expect(toJson(tree)).toMatchSnapshot();
});
