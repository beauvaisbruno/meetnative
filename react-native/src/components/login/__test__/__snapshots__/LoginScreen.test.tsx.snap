// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Button 1`] = `
<Styled(Container)>
  <Styled(Content)>
    <ErrorableImage
      source={
        Object {
          "testUri": "../../../src/components/login/img/header.png",
        }
      }
      width={750}
    />
    <Styled(ViewNB)
      padder={true}
    >
      <Connect(GoogleLoginButton) />
      <Connect(FbLoginButton) />
      <Connect(withNavigation(EmailLoginUi)) />
    </Styled(ViewNB)>
  </Styled(Content)>
</Styled(Container)>
`;
